graylog2-web-interface/src/components/cluster-configuration/data-nodes/DataNodesExpandable.tsx

```diff
@@ -0,0 +1,122 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+
+import { EntityDataTable, NoSearchResult, Spinner } from 'components/common';
+import type { ColumnSchema } from 'components/common/EntityDataTable';
+import DataNodeActions from 'components/datanode/DataNodeList/DataNodeActions';
+
+import {
+  createColumnDefinitions,
+  createColumnRenderers,
+} from './DataNodesColumnConfiguration';
+import useClusterDataNodesTableLayout from './useClusterDataNodesTableLayout';
+import useClusterDataNodes, { type ClusterDataNode } from './useClusterDataNodes';
+
+import ClusterNodesSectionWrapper from '../shared-components/ClusterNodesSectionWrapper';
+
+const EmptyState = styled(NoSearchResult)`
+  margin-top: ${({ theme }) => theme.spacings.xl};
+`;
+
+type Props = {
+  collapsible?: boolean;
+  searchQuery?: string;
+  onSelectNodeType?: () => void;
+  pageSizeLimit?: number;
+  refetchInterval?: number | false;
+};
+
+const DataNodesExpandable = ({
+  collapsible = true,
+  searchQuery = '',
+  onSelectNodeType = undefined,
+  pageSizeLimit = undefined,
+  refetchInterval = undefined,
+}: Props) => {
+  const {
+    defaultDisplayedColumns,
+    defaultColumnOrder,
+    layoutPreferences,
+    searchParams,
+    isLoadingLayout,
+    handleLayoutPreferencesChange,
+    handleSortChange,
+  } = useClusterDataNodesTableLayout(searchQuery, pageSizeLimit);
+  const {
+    nodes: dataNodes,
+    total: totalDataNodes,
+    refetch,
+    isLoading,
+    setPollingEnabled,
+  } = useClusterDataNodes(searchParams, { refetchInterval });
+
+  const columnSchemas = useMemo<Array<ColumnSchema>>(() => createColumnDefinitions(), []);
+  const columnRenderers = useMemo(() => createColumnRenderers(), []);
+
+  const handleActionsInteractionChange = useCallback(
+    (isActive: boolean) => {
+      setPollingEnabled(!isActive);
+    },
+    [setPollingEnabled],
+  );
+
+  const renderActions = useCallback(
+    (entity: ClusterDataNode) => (
+      <DataNodeActions dataNode={entity} refetch={refetch} onInteractionChange={handleActionsInteractionChange} />
+    ),
+    [handleActionsInteractionChange, refetch],
+  );
+
+  return (
+    <ClusterNodesSectionWrapper
+      title="Data Nodes"
+      titleCount={totalDataNodes}
+      onTitleCountClick={onSelectNodeType ?? null}
+      headerLeftSection={(isLoading || isLoadingLayout) && <Spinner />}
+      collapsible={collapsible}>
+      {(() => {
+        if (isLoading || isLoadingLayout) {
+          return <Spinner />;
+        }
+
+        if (dataNodes.length === 0) {
+          return <EmptyState>No Data Nodes found.</EmptyState>;
+        }
```

Comment:
It looks like these tables sometimes unmount when changing the sorting.

![Image](https://github.com/user-attachments/assets/d5e4bc84-a316-4d9a-8c95-d6da31ea7b69)